src/relation/transforms.rs

```diff
@@ -1971,4 +1989,87 @@ mod tests {
         let joined_rel = table_1.clone().cross_join(table_2.clone()).unwrap();
         joined_rel.display_dot();
     }
+
+    #[test]
+    fn test_push_grouping_columns() {
+        let table: Relation = Relation::table()
+            .name("table")
+            .schema(
+                Schema::builder()
+                    .with(("a", DataType::integer_range(1..=10)))
+                    .with(("b", DataType::integer_values([1, 2, 5, 6, 7, 8])))
+                    .with(("c", DataType::integer_range(5..=20)))
+                    .build(),
+            )
+            .build();
+
+        // no GROUP BY
+        let red = Reduce::new(
+            "reduce_relation".to_string(),
+            vec![("sum_a".to_string(), AggregateColumn::sum("a"))],
+            vec![],
+            Rc::new(table.clone()),
+        );
+        let red2 = red.clone().push_grouping_columns();
```

Comment:
what does red2 mean ? -> use meanigful naming. Naming is the first level of documentation.